packages/hydrogen/src/customer/customer.ts

```diff
@@ -584,3 +586,56 @@ function createIfInvalidCredentialThrowError(
     }
   };
 }
+
+/**
+ * This function returns a locale string in the form <language>[-<COUNTRY_CODE>], based on the provided input params.
+ * If both the i18n and the uiLocalesOverride are provided, the uiLocalesOverride will be used.
+ * If none of the params are provided, it returns null.
+ *
+ * Note: exported for testing purposes.
+ */
+export function getMaybeUILocales(params: {
+  i18n: I18nBase | null;
+  uiLocalesOverride: LanguageCode | null; // this will override i18nContext if both are provided
+}): string | null {
+  function toMaybeLocaleString(
+    language: string | null,
+    country: string | null,
+  ): string | null {
+    if (language != null) {
+      const languageLower = language.toLowerCase();
+      if (country != null) {
+        return `${languageLower}-${country.toUpperCase()}`;
+      }
+      return languageLower;
+    }
+    return null;
+  }
+
+  function uiLocalesToMaybeLocaleString(
+    uiLocales: string | null,
+  ): string | null {
+    if (uiLocales == null) {
+      return null;
+    }
+
+    // NOTE(ruggi): the locale override coming from `uiLocale` is supposed to be a LanguageCode, so it should not contain other tokens.
+    // The current implementation is kept for backwards compatibility, however we might consider adjusting it in the future.
+    const parts = uiLocales.split('-');
+    if (parts.length === 0) {
+      return null;
+    }
+
+    return toMaybeLocaleString(parts.at(0) ?? null, parts.at(1) ?? null);
+  }
```

Comment:
can these be extracted out?
(we probably don't need to recreate them every call and they could be testable that way)